src/Kubernetes.Controller/Converters/YarpParser.cs

```diff
@@ -104,6 +104,24 @@ private static void HandleIngressRulePath(YarpIngressContext ingressContext, V1S
         }
     }
 
+    private static string UpstreamName(string namespaceName, V1IngressServiceBackend ingressServiceBackend)
+    {
+        if (ingressServiceBackend != null)
+        {
+            if (ingressServiceBackend.Port.Number.HasValue && ingressServiceBackend.Port.Number.Value > 0)
+            {
+                return $"{ingressServiceBackend?.Name}.{namespaceName}:{ingressServiceBackend?.Port.Number}";
+            }
+
+            if (!string.IsNullOrWhiteSpace(ingressServiceBackend?.Name))
+            {
+                return $"{ingressServiceBackend?.Name}.{namespaceName}:{ingressServiceBackend?.Port.Name}";
+            }
```

Comment:
```suggestion
            if (ingressServiceBackend.Port.Number.HasValue && ingressServiceBackend.Port.Number.Value > 0)
            {
                return $"{ingressServiceBackend.Name}.{namespaceName}:{ingressServiceBackend.Port.Number}";
            }

            if (!string.IsNullOrWhiteSpace(ingressServiceBackend.Name))
            {
                return $"{ingressServiceBackend.Name}.{namespaceName}:{ingressServiceBackend.Port.Name}";
            }
```

`ingressServiceBackend` was already null checked above